docs/enhancements/cluster_pool_hot_spares.md

```diff
@@ -0,0 +1,162 @@
+# Maintaining Running Clusters in a ClusterPool
+
+[HIVE-1576](https://issues.redhat.com/browse/HIVE-1576)
+
+- [Maintaining Running Clusters in a ClusterPool](#maintaining-running-clusters-in-a-clusterpool)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+    - [User Stories](#user-stories)
+  - [Proposal](#proposal)
+    - [`spec.runningCount`](#specrunningcount)
+    - [Decouple PowerState on Creation](#decouple-powerstate-on-creation)
+    - [ClusterClaim Assignment by `creationTimestamp`](#clusterclaim-assignment-by-creationtimestamp)
+    - [Hibernation Controller Must Account For `hibernateAfter` in Pool Clusters](#hibernation-controller-must-account-for-hibernateafter-in-pool-clusters)
+  - [Alternatives](#alternatives)
+    - [Don't Hibernate New ClusterDeployments](#dont-hibernate-new-clusterdeployments)
+    - [`runningPercent`](#runningpercent)
+
+## Summary
+
+Support configuring ClusterPools to maintain some number of clusters (`0 <= N <= spec.size`) in Running state.
+
+## Motivation
+
+Today ClusterPools immediately hibernate new clusters.
+This means a ClusterClaim is guaranteed to suffer at least Resuming time before the cluster is available for use.
+While this is still faster than having to create a cluster from scratch, it is a potentially-unnecessary delay.
+As an example, CI runs could be reduced by ~5m if a ClusterClaim were able to grab a Running cluster rather than having to Resume a Hibernating one.
+
+### Goals
+
+Ability to tune a ClusterPool such that a ClusterClaim gets a Running cluster immediately in the common case.
+
+### Non-Goals
+
+Solving world peace.
+
+
+### User Stories
+
+1. As a consumer of a (properly tuned) ClusterPool,
+
+   I want ClusterClaim to give me a Running cluster immediately in the common case,
+
+   So that I can get my work done more quickly.
+
+   (Ideally without sacrificing the cost savings of having some clusters in the pool Hibernating.)
+
+2. As a user of a (hypothetical future) ClusterDeployment (or sibling CRD) for a cluster type that doesn't support hibernation,
+
+   I want to be able to configure a ClusterPool that never hibernates its ClusterDeployments
+
+   So that I can use ClusterPools for that kind of ClusterDeployment.
```

Comment:
hmm.. i don't think the clusterpool api mentions that the clusters it maintains have to be in hibernating state. So for a type of cd that doesn't support hibernation it can continue to maintain them as running.
so this user story imo is not necessary.

Reply:
> Punting that knowledge to the consumer via this knob would be a more flexible answer. E.g. hibernation for Zaphod clusters was added at version 3.0. Consumer configures their Zaphod 2.5 pool with runningCount==size but their 3.x pools with runningCount=0.

to me this having the user remember that size matches runningCount for cluster types that do not support hibernation is bad.
- because it allows the user to setup clusterpools that will cause errors in the hive controllers. invalid inputs should be rejected fast instead of waiting for errors. the error reporting from cluster fleet is already difficult because of noise.
- having the users know whether their configuration will support hibernation just so that the hive controller doesn't have to know that seems like bad user experience.

i personally feel that this is not a good user story. And it does not sell the runningCount feature anymore than the first use-case

Reply:
also if this usecase is that important, i think we should make changes in the proposal that would help improve the UX of size == runningCount . for example supporting percentages in addition to count will make the pool set once and forget by setting runningCount to 100%

Reply:
I've removed the user story.

However:

> having the users know whether their configuration will support hibernation just so that the hive controller doesn't have to know that seems like bad user experience.

I don't disagree; but this assumes hive *can* know. If hive *can't* know (which I contend will be the case at least some of the time) then the user is helpless.